File: in_Java/0507 Perfect Number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        if(num == 1) return false;
        int res = 1;
        for(int i = 2; i <= num/2; i++){
            if(num % i == 0) res += i;
        }
        return res == num;
    }
}

File: in_Python/0507 Perfect Number.py
class Solution(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num == 1:
            return False
        s = 1
        for i in range(2, int(sqrt(num))+1):
            if num % i == 0:
                s += i
                if i != num / i:
                    s += num / i
        return s == num